rocksdb_admin/admin_handler.cpp

```diff
@@ -440,28 +410,38 @@ void deleteTmpDBs() {
 namespace admin {
 
 AdminHandler::AdminHandler(
-  std::unique_ptr<ApplicationDBManager> db_manager,
+  std::shared_ptr<ApplicationDBManager> db_manager,
   RocksDBOptionsGeneratorType rocksdb_options): AdminHandler(
       std::move(db_manager), 
       [rocksdb_options](const std::string& dataset, const std::string& db) {
         return rocksdb_options(dataset);
   }) {}
 
 AdminHandler::AdminHandler(
-    std::unique_ptr<ApplicationDBManager> db_manager,
+    std::shared_ptr<ApplicationDBManager> db_manager,
```

Comment:
Keep this as a unique_ptr and pass as raw ptr to backup manager.

Since admin handler has ownership and backup manager is also owned by admin handler this is also okay/safe.